### PR TITLE
ddl: support `ALTER PLACEMENT POLICY ...`

### LINKED ISSUE #27714: Support ALTER PLACEMENT POLICY

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #27714

### Check List 

Tests 

- [x] Unit test

### Release note 

```release-note
ddl: support `ALTER PLACEMENT POLICY ...`
```

### CODE REVIEW

**Review by AilinKid — COMMENTED** [review]
Rest LGTM

File: ddl/placement_policy.go [hunk 1]
```diff
@@ -197,3 +197,39 @@ func onDropPlacementPolicy(t *meta.Meta, job *model.Job) (ver int64, _ error) {
...
+
+	oldPolicy, err := checkPlacementPolicyExistAndCancelNonExistJob(t, job, job.SchemaID)
+	if err != nil {
+		return ver, errors.Trace(err)
+	}
+
+	newPolicyInfo := *oldPolicy
+	newPolicyInfo.PlacementSettings = alterPolicy.PlacementSettings
+
+	err = checkPolicyValidation(&newPolicyInfo)
```
[the remark this answers is not in the document]
Comment (lcwangchao) [reply ddl/placement_policy.go:216]: ok, done

**Review by xhebox — APPROVED** [review]

**Review by AilinKid — APPROVED** [review]

**Comment by AilinKid** [comment]
> /run-check_dev_2

[2021-09-03T09:16:22.657Z] placement_policy_test.go:222:

[2021-09-03T09:16:22.657Z]     c.Assert(err.Error(), Equals, ca.errmsg)

[2021-09-03T09:16:22.657Z] ... obtained string = "" +

[2021-09-03T09:16:22.657Z] ...     "invalid label constraints format: should be [constraint1, ...] (error yaml: unmarshal errors:\n" +

[2021-09-03T09:16:22.657Z] ...     "  line 1: cannot unmarshal !!map into []string)"

[2021-09-03T09:16:22.657Z] ... expected string = "" +

[2021-09-03T09:16:22.657Z] ...     "[ddl:-1]invalid label constraints format: should be [constraint1, ...] (error yaml: unmarshal errors:\n" +

[2021-09-03T09:16:22.657Z] ...     "  line 1: cannot unmarshal !!map into []string)"

**Pull request merged**